### ISSUE TITLE
There are unnecessary options in the .env.sample for self-hosting

### ISSUE THREAD

**Issue by XIYO** [supabase/supabase#26578]
# Improve documentation

## Link 

[/docker/.env.sample](https://github.com/supabase/supabase/blob/master/docker/.env.example)

## Describe the problem

Self-hosting does not support the use of Studio. However, the sample includes this option.

## Describe the improvement

Remove settings related to Studio.

## Additional context

Since Studio cannot be used in self-hosting, it is necessary to remove the description to prevent confusion among users. Alternatively, a detailed explanation should be provided.

```diff
## General
-SITE_URL=http://localhost:3000
ADDITIONAL_REDIRECT_URLS=
JWT_EXPIRY=3600
DISABLE_SIGNUP=false
API_EXTERNAL_URL=http://localhost

...


-############
-# Studio - Configuration for the Dashboard
-############

-STUDIO_DEFAULT_ORGANIZATION=Default Organization
-STUDIO_DEFAULT_PROJECT=Default Project

-STUDIO_PORT=3000
```

**Comment by cyluss** [comment]
Self hosting **does support** Studio but without hosting functionality such as project management. What leads you to conclude that the self-hosted version does not include Studio?
![image](https://github.com/supabase/supabase/assets/41672371/4f3bb2ad-c564-4448-a24a-3315c4eba5ff)

**Comment by XIYO** [comment]
In the provided Docker Compose example, I noticed that the studio has no exposed ports. When I force the connection through port 3000, it ultimately redirects to the same screen as the web interface on port 8000, which is specified under kong.

I understood that the studio is a tool for creating and managing multiple instances (with self-hosting not supporting multi-instance management), and kong is a tool for using these instances.

If the studio refers to a web UI, then I have misunderstood its concept. The Docker Compose configuration seems somewhat confusing for first-time users.

The options related to the studio mentioned above intrigue me and prompt experimentation. However, it is not documented that self-hosting does not support the studio (here referring to the multi-instance management functionality). This information can only be inferred from discussions, indicating that the feature's documentation is not clear.

#4907

**Comment by cyluss** [comment]
I understood. Currently one docker-compose.yaml deploys one Supabase instance, not management plane of Supabase.

I guess [Self hosting guide](https://supabase.com/docs/guides/self-hosting/docker) can be improved in two ways:
1. Present deployment diagram: to clarify what resources will reader get after deployment
2. Clarify the scope of self-hosted dashboard: Self-hosted dashboard is deployed at instance level, which implies the following.
  - User cannot perform deployment tasks using Dashboard
  - Each Supabase instances must be deployed separately
  
[Studio readme](https://github.com/supabase/supabase/tree/429f3d2/apps/studio#whats-included) does mention point 2, but it is better to notice readers the limitations upfront instead of expecting them to do their own experiments.

**Comment by yuqianma** [comment]
I ran into this misunderstanding too.

Also, there is a `SITE_URL` key in the `.env` that defaults to `http://localhost:3000`, which adds to the confusion.

If we can, it would be great to have an "example config" section in the doc. We could use "mydomain.com" as an example for how users can set up their own domain.

PS:
I'm new to docker-compose things. But after I found out that `:8000` opens the studio, I took a look at  the`docker-compose.yml`, and noticed some configs in `./volumes/api/kong.yml`. It shows `/* -> http://studio:3000/*` in the last section. So all services are organized by kong. Hopefully, this info helps anyone else who's new to docker-compose too!

**Comment by cyluss** [comment]
@yuqianma I agree, document assumes local deployment with Docker compose and Supabase on the user's PC. To improve, the guide could explain remote deployment, for example domain name and external IP for VM. This would cover basic self hosting scenario served by a single machine.

Here's my suggestions:

1. Clarify the goal of guide as remote machine deployment
2. Add comment for SITE_URL `.env.sample` for domain setup

**Comment by cyluss** [comment]
#29240

**Comment by Hallidayo** [comment]
Hi all, We’re going to close this one due to inactivity. Please feel free to reopen if you’re still having issues here or open a new issue.
